### PR TITLE
Definition json to yaml

### PR DESCRIPTION
**Pull request by anatbal**
## Description
- Change the definition.json into definiton.yml
- Refactor validation workflow
- Refactor TF workflow to extract from yml instead of json

### Checklist:

- [ ] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -14,17 +13,29 @@ jobs:
...
       - name: Validate JSON
-        if: ${{ steps.calc_files_diff.outputs.diff != '' }}
+        # if: ${{ steps.calc_files_diff.outputs.diff != '' }}
+        if: always()
         uses: docker://orrosenblatt/validate-json-action:latest
         env:
           INPUT_SCHEMA: .ci/trial-json-schema.json
-          INPUT_JSONS: ${{ steps.calc_files_diff.outputs.diff }}
+          # INPUT_JSONS: ${{ steps.calc_files_diff.outputs.diff }}
+          INPUT_JSONS: sample_trial/definition.json
```
Comment (eladiw) [inline .github/workflows/validate.yml]: path does not exist

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -14,17 +13,29 @@ jobs:
...
+      
+      - name: Validate YML
+        if: always()
+        uses: nrkno/yaml-schema-validator-github-action@master
+        with:
+          schema: '.ci/trial-json-schema.json'
+          target: 'sample_trial/definition.yml'
+          # Uncomment to enable strict checks
```
Comment (eladiw) [inline .github/workflows/validate.yml]: if we can lets enable. otherwise remove these two lines

File: .github/workflows/validate.yml [hunk 3]
```diff
@@ -14,17 +13,29 @@ jobs:
...
 
       - name: Validate JSON
-        if: ${{ steps.calc_files_diff.outputs.diff != '' }}
+        # if: ${{ steps.calc_files_diff.outputs.diff != '' }}
+        if: always()
         uses: docker://orrosenblatt/validate-json-action:latest
         env:
           INPUT_SCHEMA: .ci/trial-json-schema.json
-          INPUT_JSONS: ${{ steps.calc_files_diff.outputs.diff }}
+          # INPUT_JSONS: ${{ steps.calc_files_diff.outputs.diff }}
```
Comment (eladiw) [inline .github/workflows/validate.yml]: why commented

File: .github/workflows/validate.yml (hunk 1 above)
Comment (anatbal) [reply .github/workflows/validate.yml]: Resolved

File: .github/workflows/validate.yml (hunk 2 above)
Comment (anatbal) [reply .github/workflows/validate.yml]: Resolved, earlier version

File: .github/workflows/validate.yml (hunk 3 above)
Comment (anatbal) [reply .github/workflows/validate.yml]: Resolved, earlier version

**Review by eladiw — DISMISSED** [review]

File: .github/workflows/validate.yml [hunk 4]
```diff
@@ -1,8 +1,7 @@
-name: ValidateTrialJsonSchema
+name: Validate Trial Config Files
```
Comment (tamirkamara) [inline .github/workflows/validate.yml]: I think this is a good time to move the validation step to the primary workflow since we don't want to deploy if the validation fails.
Also, I think we can skip the hassle of figuring which definition changed and just validate all of them.

File: .github/workflows/validate.yml [hunk 5]
```diff
@@ -14,17 +13,24 @@ jobs:
...
           echo "Files changed since last commit..."
           echo "$FILES"
 
           echo "::set-output name=diff::$( echo "$FILES" | tr '\n' ',' | sed '$s/,$/\n/' )"
 
-      - name: Validate JSON
-        if: ${{ steps.calc_files_diff.outputs.diff != '' }}
-        uses: docker://orrosenblatt/validate-json-action:latest
+      - name: Validate YML
+        if: always()
```
Comment (tamirkamara) [inline .github/workflows/validate.yml]: why always?

File: .github/workflows/terraform-plan-apply.yml [hunk 6]
```diff
@@ -12,6 +12,7 @@ jobs:
       is_empty_matrix: ${{ steps.set-matrix.outputs.is_empty_matrix }}
     steps:
       - uses: actions/checkout@v2
+      - uses: chrisdickinson/setup-yq@latest
```
Comment (tamirkamara) [inline .github/workflows/terraform-plan-apply.yml]: Is this the latest version of YQ? It doesn't look like it... also, the syntax of the command is a bit weird...
Are you sure we should use it (hardly any reviews on the action)?

**Review by tamirkamara — CHANGES_REQUESTED** [review]

File: .github/workflows/validate.yml (hunk 4 above)
Comment (anatbal) [reply .github/workflows/validate.yml]: Agree, done

File: .github/workflows/validate.yml (hunk 5 above)
Comment (anatbal) [reply .github/workflows/validate.yml]: fixed

File: .github/workflows/terraform-plan-apply.yml (hunk 6 above)
Comment (anatbal) [reply .github/workflows/terraform-plan-apply.yml]: Fixed, after the fix didn't change jq syntax besides replacing -r to eval

File: .github/workflows/terraform-plan-apply.yml [hunk 7]
```diff
@@ -39,13 +41,23 @@ jobs:
...
+        id: validate_definiton_yml
+        env:
+          INPUT_SCHEMA: .ci/trial-json-schema.json
+        run: |
+          sudo npm install -g ajv-cli
+          export FILES=$(find . -name 'definition.yml')
+          for FILE in $(echo "$FILES" | sed "s/,/ /g")
+          do
+               ajv validate -s $INPUT_SCHEMA -d "$FILE"
+          done
```
Comment (tamirkamara) [inline .github/workflows/terraform-plan-apply.yml]: This can be written with the exec option of find:
https://stackoverflow.com/questions/9612090/how-to-loop-through-file-names-returned-by-find

File: .github/workflows/terraform-plan-apply.yml [hunk 8]
```diff
@@ -21,12 +26,9 @@ jobs:
           set -o pipefail
           set -o nounset
           # set -o xtrace
-
```
Comment (tamirkamara) [inline .github/workflows/terraform-plan-apply.yml:24]: Why did you delete all empty lines?

**Review by tamirkamara — CHANGES_REQUESTED** [review]

File: .github/workflows/terraform-plan-apply.yml (hunk 7 above)
Comment (anatbal) [reply .github/workflows/terraform-plan-apply.yml]: Fixed

File: .github/workflows/terraform-plan-apply.yml (hunk 8 above)
Comment (anatbal) [reply .github/workflows/terraform-plan-apply.yml:24]: Mistake caused by auto reformat on VS code... fixed

File: .github/workflows/terraform-plan-apply.yml [hunk 9]
```diff
@@ -202,15 +220,15 @@ jobs:
 
               var_name=${line_array[0]}
               json_path=${line_array[1]}
-              json_value=$(jq -r ${json_path} ${{ matrix.definition_path }})
+              json_value=$(yq eval ${json_path} ${{ matrix.definition_path }})
```
Comment (tamirkamara) [inline .github/workflows/terraform-plan-apply.yml]: can you change from json to something else?
property_path / property_value?

**Review by tamirkamara — CHANGES_REQUESTED** [review]

File: .github/workflows/terraform-plan-apply.yml (hunk 9 above)
Comment (anatbal) [reply .github/workflows/terraform-plan-apply.yml]: Done

**Review by tamirkamara — APPROVED** [review]

**Review by eladiw — APPROVED** [review]
